typings/downshift.flow.js

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+
+// @flow
+
+type A11StatusMessageOptions = {
+  highlightedIndex: number,
+  highlightedValue: any,
+  inputValue: string,
+  isOpen: boolean,
+  itemToString: (item: any) => string,
+  previousResultCount: number,
+  resultCount: number,
+  selectedItem: any,
+}
+
+type ChangeOptions = {
+  selectedItem: any,
+  previousItem: any,
+}
+
+type StateChangeOptions = {
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type GetRootPropsOptions = {
+  refKey: string,
+}
+
+type GetItemPropsOptions = {
+  index: number,
+  item: any,
+}
+
+type ControllerStateAndHelpers = {
+  // prop getters
+  getRootProps: (options: GetRootPropsOptions) => any,
+  getButtonProps: (options: GetButtonPropsOptions) => any,
+  getLabelProps: (options: GetLabelPropsOptions) => any,
+  getInputProps: (options: GetInputPropsOptions) => any,
+  getItemProps: (options: GetItemPropsOptions) => any,
+
+  // actions
+  openMenu: () => void,
+  closeMenu: () => void,
+  toggleMenu: () => void,
+  selectItem: (item: any) => void,
+  selectItemAtIndex: (index: number) => void,
+  selectHighlightedItem: (index: number) => void,
```

Comment:
This property def is incorrect. It takes an optional `otherStateToSet` object and optional callback function.